### ISSUE TITLE
Use a consistent notation for required/optional fields

### ISSUE THREAD

**Issue by cehsu** [usds/us-forms-system#297]
**Is your feature request related to a problem? Please describe.**
The design system's errorable checkbox component has been [updated](https://github.com/department-of-veterans-affairs/design-system/pull/166) to allow the `required` style of the privacy agreement to be consistent with the form fields.
![image](https://user-images.githubusercontent.com/16051172/46639898-c4243f00-cb1d-11e8-900c-15f7c62656f7.png)

**Describe the solution you'd like**
Use latest version of @department-of-veterans-affairs/formation

**Additional context**
Here is a screenshot of the current privacy agreement component:
![localhost_3001_education_opt-out-information-sharing_opt-out-form-0993_applicant_information 11](https://user-images.githubusercontent.com/16051172/42600479-511bfeea-8517-11e8-8365-80468e7d107a.png)

**Comment by dmethvin-gov** [comment]
We removed the direct dependency on formation in #123 so I guess for 1.x-stable we would need to update our copy of `ErrorableCheckbox` and other adopted widgets to match. For 2.x and later we should determine a solution that doesn't require these kind of changes since we've basically forked off formation. 

Adding the English word `Required` to the field complicates the i18n process and that's something USFS will need to tackle sooner rather than later. I agree that a simple `*` doesn't convey required-ness though so I understand why it was changed. This seems like something that library users will have an opinion about so it will need to be configurable.

**Comment by cehsu** [comment]
Should I set up a PR for the first option, to update the `ErrorableCheckbox` component?

**Comment by dmethvin-gov** [comment]
It looks like titles/labels are implemented in [react-jsonschema-form/SchemaField](https://github.com/department-of-veterans-affairs/react-jsonschema-form/blob/master/src/components/fields/SchemaField.js#L42) as well, using only (`*`).

**Comment by annekainicUSDS** [comment]
@cehsu In digging into this, I noticed that USWDS has [moved away from indicating required fields](https://designsystem.digital.gov/components/form-templates/), and instead indicates the optional ones. In general I think we want to adhere to USWDS, unless we have a strong reason not to. Is there a reason why Formation has veered from USWDS's guidance? Is this just because we're continuing to mark required fields as we have in the past, or is there a specific reason? 

I'd like to hold off on making any code changes until we know the direction we want to go in and why, and right now I think the problem is beyond just a code change, but a UX question.

**Comment by cehsu** [comment]
Thanks @dmethvin-gov @annekainicUSDS, this is in the interest of consistency and clarity, @peggygannon can provide additional insights.

**Comment by dmethvin-gov** [comment]
As a somewhat-related aside, I've done a deep-dive into the [Form 1990](https://www.vets.gov/education/apply-for-education-benefits/application/1990/introduction) and it seems like many of the questions deeper in the form are optional when they should be required. The yes/no questions about ROTC and whether they're on active duty are examples.

**Comment by annekainicUSDS** [comment]
I'm pretty sure those fields are not actually required from a business perspective in order to process the form; curious what you're seeing that indicates that they are incorrectly marked?

**Comment by dmethvin-gov** [comment]
It just seems odd to have questions that are Yes/No that don't need to be answered, especially if they have dependent information. If answers for those questions really aren't needed it would be good to leave it out to make the form less complex.

**Comment by kfrz** [comment]
Bumping this conversation. 

While digging into https://github.com/department-of-veterans-affairs/vets.gov-team/issues/13828 I ran into the above discussion. I'm wondering if there's anything actionable here so we can plan for prioritization.

**Comment by annekainicUSDS** [comment]
@kfrz We're waiting on the Vets.gov team to provide feedback on the decision to keep (*Required) indications as opposed to moving towards marking optional fields, as USWDS is now recommending.

**Comment by cehsu** [comment]
Thanks @annekainicUSDS, for the time being we are interested in making instances of required field labels consistent (replacing `*` with `*Required`). The question of whether optional fields should be labeled instead is a question that is being discussed as well. Would it make sense to set up another ticket for that conversation?

**Comment by dmethvin-gov** [comment]
I think I may have raised some false hope here. A changed to add `*Required` *everywhere* also means that `react-jsonschema-form` needs to be updated, so it's not completely under the control of the library. It seems unusual to update rjsf for this as well, at least to me but it would be useful to hear other perspectives. The minimum-distance fix to get consistency is to change `FieldTemplate` which is currently the only one using `*Required`.

**Comment by jbalboni** [comment]
@dmethvin-gov @annekainicUSDS If we wanted to submit a PR to update CheckBoxWidget and ErrorableCheckbox to use `(*Required)` on the 1.x branch, would that be ok? The two instances in RJSF mentioned above I don't think ever show up in the VA.gov forms.

**Comment by annekainicUSDS** [comment]
@jbalboni I think that would be okay for now; it sounds like we need to do a more thorough review of how these components are built and used between rjsf and USFS. @dmethvin-gov are you okay with this change for now?

**Comment by dmethvin-gov** [comment]
Fixed in 1.x-stable with 2b44328ec265a17f78d150ff07fce57a96e437b9
